lib/associations/belongs-to-many.js

```diff
@@ -562,7 +562,7 @@ class BelongsToMany extends Association {
         const where = {};
         where[identifier] = sourceInstance.get(sourceKey);
         where[foreignIdentifier] = obsoleteAssociations.map(obsoleteAssociation => obsoleteAssociation[foreignIdentifier]);
-
+        _.assign(where, association.through.scope);
```

Comment:
This handles destroy case, We also need to handle the other case when associations are updating and a test as well